Factor vector intrinsics out of SumKernel.cpp

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61083 Add BFloat16 support to CPU nansum
* **#61483 Factor vector intrinsics out of SumKernel.cpp**
* #61082 Use cascade-summation to improve nansum accuracy

This will make it simpler to support AVX512 which is upcoming in #56992, see https://github.com/pytorch/pytorch/pull/56992#discussion_r667060280 for reference.

Differential Revision: [D29753536](https://our.internmc.facebook.com/intern/diff/D29753536)

peterbell10 committed:
Factor vector intrinsics out of SumKernel.cpp

[ghstack-poisoned]

peterbell10 committed:
Update on "Factor vector intrinsics out of SumKernel.cpp"


This will make it simpler to support AVX512 which is upcoming in #56992, see https://github.com/pytorch/pytorch/pull/56992#discussion_r667060280 for reference.

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29753536).

peterbell10 committed:
Update on "Factor vector intrinsics out of SumKernel.cpp"


This will make it simpler to support AVX512 which is upcoming in #56992, see https://github.com/pytorch/pytorch/pull/56992#discussion_r667060280 for reference.

Differential Revision: [D29753536](https://our.internmc.facebook.com/intern/diff/D29753536)

[ghstack-poisoned]
